test/e2e/apps/statefulset.go

```diff
@@ -510,6 +512,46 @@ var _ = SIGDescribe("StatefulSet", func() {
 
 		})
 
+		ginkgo.It("should perform canary updates and phased rolling updates of template modifications for partiton1 and delete pod-0 without failing container", func(ctx context.Context) {
+			ginkgo.By("Creating a new StatefulSet without failing container")
+			ss := e2estatefulset.NewStatefulSet("ss2", ns, headlessSvcName, 3, nil, nil, labels)
+			deletingPodForRollingUpdatePartitionTest(ctx, f, c, ns, ss)
+		})
+
+		ginkgo.It("should perform canary updates and phased rolling updates of template modifications for partiton1 and delete pod-0 with failing container", func(ctx context.Context) {
+			ginkgo.By("Creating a new StatefulSet with failing container")
+			ss1 := e2estatefulset.NewStatefulSet("ss3", ns, headlessSvcName, 3, nil, nil, labels)
+			ss1.Spec.Template.Spec.Containers = append(ss.Spec.Template.Spec.Containers, v1.Container{
+				Name:    "sleep-exit-with-1",
+				Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+				Command: []string{"sh", "-c"},
+				Args: []string{`if [ "${POD_NAME}" = "ss3-0" ]; then
+									_term(){
+										echo "Received SIGTERM signal"
+										exit 1
+									}
```

Comment:
Maybe we could do the `if` inside the trap to just exit with 0 or 1 depending on the pod name. I think in the current form the container for pod-1 and pod-2 will continuously get restarted as it exits with 0 (since restartPolicy=Always).

Reply:
I have fixed it now it should be fine. Thank you for the help.

Reply:
Thanks, feel free to mark resolved.